app/code/core/Mage/Catalog/Helper/Category/Url/Rewrite.php

```diff
@@ -95,7 +95,7 @@ public function joinTableToCollection(Mage_Catalog_Model_Resource_Category_Flat_
             'url_rewrite.category_id = main_table.entity_id AND url_rewrite.is_system = 1 '.
                 ' AND ' . $collection->getConnection()->quoteInto('url_rewrite.store_id = ?', $storeId).
                 ' AND url_rewrite.category_id IS NOT NULL'.
```

Comment:
Is there ever a time when `url_rewrite.category_id IS NOT NULL` is true and `url_rewrite.url_type` is not "category"?